src/Compiler/pars.fsy

```diff
@@ -1506,26 +1506,52 @@ attributeListElements:
 
 
 /* One custom attribute */
-attribute:
+attribute:     
```

Comment:
Would it be possible to add recovery for unfinished attributes? For example, consider typing a new attribute and then adding type arguments to it. We want the parser to be able to parse the rest of the file while the attribute is being typed, if that is possible with the language grammar.

```fsharp
[<Attr<>]
()
```

```fsharp
[<Attr<>>]
()
```

```fsharp
[<Attr<arg>]
()
```

```fsharp
[<Attr<_>]
()
```

```fsharp
[<Attr<arg; Attr2>]
()
```

```fsharp
[<Attr<(LiteralArg)>]
()
```

```fsharp
[<Attr< LiteralArg>]
()
```

Reply:
I'm not entirely certain how this would be done at the moment, but I will begin playing the idea now.

There is a complication with the lexer and typars being immediately adjacent to a closing `>]` that I think might end up being an obstacle in this case - I include in case I am missing something and someone can enlighten me on the inner workings on the lexer:
In order to allow for the closing `>]` to be immediately adjacent to a typar, e.g.:
```fsharp
[<Attr<_>>]
```
We need to deal with the fact that the lexer is going to collect the whole series of greater-thans. Based on my current understanding of how the lexer works, I think the only place we can deal with this is in the `LexFilterImpl.peekAdjacentTypars` function in `Compiler\SyntaxTree\LexFilter.fs`, in particular when the stack is being constructed via scan ahead (I assume the scan ahead section is the best place as we need to know that a `]` is immediately adjacent in order to be sure we can smash the last gt off the end of what the lexer collects).